langmeta/langmeta/shared/src/main/scala/org/langmeta/semanticdb/internal/package.scala

```diff
@@ -144,102 +204,159 @@ package object semanticdb {
+                if (dtest(d.CONTRAVARIANT)) sflip(s.SymbolInformation.Property.CONTRAVARIANT.value)
+                sproperties
+              }
+              val sname = ddenot.name
+              val srange = None
+              val ssignature = {
+                val stext = ddenot.signature
+                val soccurrences = ddenot.names.map {
+                  case d.ResolvedName(dpos: org.langmeta.Position.Range, ssym, disDefinition(srole)) =>
+                    val srange = s.Range(dpos.startLine, dpos.startColumn, dpos.endLine, dpos.endColumn)
+                    s.SymbolOccurrence(Some(srange), ssym.syntax, srole)
+                  case other =>
+                    sys.error(s"bad database: unsupported name $other")
+                }
+                Some(s.TextDocument(text = stext, occurrences = soccurrences))
               }
-              val smembers = ddefn.members.map(_.syntax)
-              val soverrides = ddefn.overrides.map(_.syntax)
-              Some(s.Denotation(flags, name, signature, snames, smembers, soverrides))
+              val smembers = ddenot.members.map(_.syntax)
+              val soverrides = ddenot.overrides.map(_.syntax)
+              Some(s.SymbolInformation(ssymbol, slanguage, skind, sproperties, sname, srange, ssignature, smembers, soverrides))
             }
           }
           object dSynthetic {
             def unapply(dsynthetic: dSynthetic): Option[s.Synthetic] = dsynthetic match {
-              case d.Synthetic(dPosition(spos), ssyntax, dnames) =>
-                val snames = dnames.toIterator.map {
-                  case d.ResolvedName(org.langmeta.inputs.Position.Range(_, sstart, send), ssym, sisDefinition) =>
-                    s.ResolvedName(Some(s.Position(sstart, send)), ssym.syntax, sisDefinition)
-                  case other =>
-                    sys.error(s"bad database: unsupported name $other")
-                }.toSeq
-                Some(s.Synthetic(Some(spos), ssyntax, snames))
+              case d.Synthetic(dPosition(srange), ssyntax, dnames) =>
+                val stext = {
+                  val stext = ssyntax
+                  val soccurrences = dnames.toIterator.map {
+                    case d.ResolvedName(dpos: org.langmeta.Position.Range, ssym, disDefinition(srole)) =>
+                      val srange = s.Range(dpos.startLine, dpos.startColumn, dpos.endLine, dpos.endColumn)
+                      s.SymbolOccurrence(Some(srange), ssym.syntax, srole)
+                    case other =>
+                      sys.error(s"bad database: unsupported name $other")
+                  }.toSeq
+                  Some(s.TextDocument(text = stext, occurrences = soccurrences))
+                }
+                Some(s.Synthetic(Some(srange), stext))
               case _ =>
                 None
             }
           }
-          val (splatformpath, scontents) = dinput match {
+          val sformat = "semanticdb3"
```

Comment:
Can be bind this magic variable somewhere globally for reuse?

Reply:
Done.